### PR TITLE
Change password

### PR DESCRIPTION
**Pull request by mato2593**
This PR includes the view implementation of the reset password dialog, the reset password service call implementation and checks if the user is from facebook or not to show the password field in the user profile. 

https://trello.com/c/XJtd86mC/11-3-as-a-user-i-should-be-able-to-reset-my-password

change-password-errors

reset-password-successful

### CODE REVIEW

File: Target-MVD-Mathias/Networking/Services/UserAPI.swift [hunk 1]
```diff
@@ -137,6 +136,25 @@ class UserAPI {
     }
   }
   
+  class func resetPassword(_ previousPassword: String, newPassword: String, success: @escaping () -> Void, failure: @escaping (_ error: Error) -> Void) {
+    let url = usersUrl + "\(UserDataManager.getUserId())" + "/password/change"
```
Comment (fernandatoledo) [inline Target-MVD-Mathias/Networking/Services/UserAPI.swift]: You can avoid that last concat by doing `let url = usersUrl + "\(UserDataManager.getUserId())/password/change"`

**Review by mjurfest — APPROVED** [review]

File: Target-MVD-Mathias/ViewControllers/UserProfileViewController.swift [hunk 2]
```diff
@@ -35,6 +49,30 @@ class UserProfileViewController: UIViewController {
   var username = ""
   var email = ""
   
+  lazy var goBackToHomeNavigationItem: UIBarButtonItem = {
+    return UIBarButtonItem(image: #imageLiteral(resourceName: "ForwardArrow"), style: .plain, target: self, action:#selector(UserProfileViewController.goBackToHome))
```
Comment (fernandatoledo) [inline Target-MVD-Mathias/ViewControllers/UserProfileViewController.swift]: MIssing space after action:

Comment (pMalvasio) [inline Target-MVD-Mathias/ViewControllers/UserProfileViewController.swift]: You can use `goBackToHome ` without the controller inside the selector

File: Target-MVD-Mathias/Networking/Services/UserAPI.swift (hunk 1 above)
Comment (mato2593) [reply Target-MVD-Mathias/Networking/Services/UserAPI.swift]: Copy paste issues 🤦‍♂️

File: Target-MVD-Mathias/Manager/UserDataManager.swift [hunk 3]
```diff
@@ -39,6 +39,10 @@ class UserDataManager: NSObject {
     return self.getUserObject()?.id ?? 0
   }
   
+  class func isUserFromFacebook() -> Bool {
```
Comment (pMalvasio) [inline Target-MVD-Mathias/Manager/UserDataManager.swift]: Maybe just a simple key for this that you can set inside the login with Facebook method and thats it. WDYT?

Comment (mato2593) [reply Target-MVD-Mathias/Manager/UserDataManager.swift]: Yeah, I think that approach is better, I'll change it

**Review by pMalvasio — APPROVED** [review]

**Review by fernandatoledo — APPROVED** [review]

**Pull request merged**